### ISSUE TITLE
Poor performance on othello

### ISSUE THREAD

**Issue by jiacheng-ye** [deepmind/open_spiel#1099]
Hi, thanks for your brilliant repo. I'm trying to training a alphazero model on othello. 

To begin with, I first tried the example of tictactoe with the default parameters in `tic_tac_toe_alpha_zero.py` and the result is good, i.e., almost draw with MCTS (25*1000 simulation by set `eval_level=7`, exhausted seach for tictactoe if I understand correctly).

Now for othello, I guess the task becames harder so I changed the resnet blocks from 2 to 6. However, it can hardly beat MCTS with eval_level=7. Here is the final evaluator log in tictactoe and othello:
```
# othello
AZ: -1.0, MCTS: 1.0, AZ avg/50: -0.960

# tictatoe
AZ: 0.0, MCTS: 0.0, AZ avg/50: 0.080
```

The loss seems dropped at a similar magnitude (i.e., from 3.32 to 2.65) with that in tictactoe (i.e., from 2.17 to 1.62).

What can be a possible reason and how to tune the parameters for harder tasks like othello? Any advice would be helpful. 

Thanks.

**Comment by lanctot** [comment]
How long are you letting it train for? Are you using convolutional neural nets?

**Comment by lanctot** [comment]
Is it beating random, or MCTS at lower simulation counts?

**Comment by jiacheng-ye** [comment]
Thanks for your reply. This is the used config.
```
{
  "actors": 4,
  "checkpoint_freq": 50,
  "eval_levels": 7,
  "evaluation_window": 50,
  "evaluators": 4,
  "game": "othello",
  "learning_rate": 0.01,
  "max_simulations": 20,
  "max_steps": 50,
  "nn_depth": 6,
  "nn_model": "resnet",
  "nn_width": 128,
  "observation_shape": [
    3,
    8,
    8
  ],
  "output_size": 65,
  "path": "outputs/othello/",
  "policy_alpha": 0.25,
  "policy_epsilon": 1,
  "quiet": true,
  "replay_buffer_reuse": 4,
  "replay_buffer_size": 16384,
  "temperature": 1,
  "temperature_drop": 10,
  "train_batch_size": 128,
  "uct_c": 1,
  "weight_decay": 0.0001
}
```


> Is it beating random, or MCTS at lower simulation counts?

I just tried on 5 games:
- az:random = 5:0. 
- az(sim=1000):mcts(sim=1000)=0:5
- az(sim=10):mcts(sim=10)=0:5

It seems the model not learns much from mcts self-play :(  

Also, I tried to set `max_simulations` to 100 during training. The results still similar.

Maybe I should use a larger model, say 20 blocks?

BTW, is there any command for two-palyer playing in parallel (like the parallel self-play in the training time is controlled by set multiple actors), so that I can let two system (e.g., nn vs.nn or nn vs. mcts) combat more efficiently?

**Comment by lanctot** [comment]
How many training steps? 

With only 3 hours with a basic implementation like ours is almost surely not enough. Please compare the number of training steps with the AlphaZero paper. Othello is quite a large game, you might need something similar to Go.

20 simulations is too low. 100 is probably also too low.

Also, I believe @tewalds tried training it on Hex and had some graphs for that at some point. Might be worthwhile checking with him.

**Comment by jiacheng-ye** [comment]
> How many training steps?

I use 50 steps. 

>Please compare the number of training steps with the AlphaZero paper. Othello is quite a large game, you might need something similar to Go. 20 simulations is too low. 100 is probably also too low.

Thanks for your suggestion. I will have a try.

> Also, I believe @tewalds tried training it on Hex and had some graphs for that at some point. Might be worthwhile checking with him.

Cool. @tewalds, would you kindly give some advice on this point? Thanks for your help.

**Comment by tewalds** [comment]
I didn't always get great performance either, with some mode collapse where the first player made a move and the second player immediately gave up. I suspect I would need some forced random openings to get diversity of gameplay, but didn't implement that. 

https://github.com/deepmind/open_spiel/blob/master/open_spiel/python/algorithms/alpha_zero/analysis.py generates the graphs that @lanctot was talking about.

**Comment by tewalds** [comment]
Here's one large training run I did on havannah. It needed ~250 training steps to beat MCTS with 1000x more rollouts 80% of the time. Different games take a different number of steps, and it depends on your exploration constant amoung other things.
![image](https://github.com/deepmind/open_spiel/assets/256625/c5834acf-8992-4918-9d66-c23aaa007794)
